sway/input/cursor.c

```diff
@@ -37,6 +37,28 @@ static uint32_t get_current_time_msec(void) {
 	return now.tv_sec * 1000 + now.tv_nsec / 1000000;
 }
 
+static struct wlr_surface *input_popup_surface_at(struct sway_output *output,
+		struct sway_input_method_relay *relay, double ox, double oy, double *sx, double *sy) {
+	struct sway_input_popup *popup;
+	wl_list_for_each(popup, &relay->input_popups, link) {
+		int lx, ly;
+		if (!sway_input_popup_get_position(popup, &lx, &ly)) {
+			continue;
+		}
+		if (!popup->popup_surface->mapped || !popup->visible) {
+			continue;
+		}
+		double _sx = ox - lx;
+		double _sy = oy - ly;
```

Comment:
This mixes up layout-local coords and output-local coords. I think `input_popup_surface_at` should take layout-local coords instead.